src/3p-frame.js

```diff
@@ -212,22 +212,30 @@ export function resetBootstrapBaseUrlForTesting(win) {
  * @return {string}
  */
 export function getDefaultBootstrapBaseUrl(parentWindow, opt_srcFileBasename) {
-  if (parentWindow.defaultBootstrapBaseUrl) {
-    return parentWindow.defaultBootstrapBaseUrl;
-  }
   const srcFileBasename = opt_srcFileBasename || 'frame';
+  // Cached at filename level to ensure different url is returned.
+  parentWindow.defaultBootstrapBaseUrl =
+      parentWindow.defaultBootstrapBaseUrl || {};
+  if (parentWindow.defaultBootstrapBaseUrl[srcFileBasename]) {
+    return parentWindow.defaultBootstrapBaseUrl[srcFileBasename];
+  }
   if (getMode().localDev || getMode().test) {
     if (overrideBootstrapBaseUrl) {
-      return parentWindow.defaultBootstrapBaseUrl = overrideBootstrapBaseUrl;
+      return parentWindow.defaultBootstrapBaseUrl[srcFileBasename] =
+          overrideBootstrapBaseUrl;
     }
-    return parentWindow.defaultBootstrapBaseUrl = getAdsLocalhost(parentWindow)
-        + '/dist.3p/'
-        + (getMode().minified ? `$internalRuntimeVersion$/${srcFileBasename}`
-            : `current/${srcFileBasename}.max`)
-        + '.html';
+    return parentWindow.defaultBootstrapBaseUrl[srcFileBasename] =
```

Comment:
I see that the first part of the url are all the same. Can we cached the first part instead, and then attached `${srcFileBasename}` to the cached url.

Reply:
Yah, constructing a string is easy, we just need the cached domain to always be the same.

Reply:
PTAL - changed so that only subdomain is cached